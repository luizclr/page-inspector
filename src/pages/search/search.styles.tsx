import styled from "styled-components";

export const Title = styled.h2`
  font-size: 18pt;
  margin-bottom: 20px;
  margin-left: 30px;
  color: #474b4e;
`;

export const ReloadButton = styled.button`
  margin: 15px;
  font-size: 12pt;
  padding: 10px;
  border: none;
  border-radius: 5px;
  border: 1px solid #5dc460;
  background-color: #fff;
  color: #42ab49;
  font-weight: bold;

  :hover {
    background-color: #5dc460;
    color: #fff;
  }
`;

export const ResultListWrapper = styled.div`
  width: 100%;
`;

export const ResultList = styled.ul`
  list-style-type: none;
  max-height: 200px;
  overflow-y: auto;
  padding-left: 15px;
`;

export const ResultListItem = styled.li`
  padding: 5px;
`;
