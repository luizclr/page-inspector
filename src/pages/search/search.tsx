import { SingleInputForm } from "~/components/single-input-form/single-input-form";
import { ResultList, ResultListItem, Title } from "~/pages/search/search.styles";
import { useInspection } from "~/providers/inspection/inspection";
import { InspectionActionTypes } from "~/reducers/inspector/types";
import { isNil } from "~/utils";

export const Search: React.FC = () => {
  const { inspection, inspectionService, dispatch } = useInspection();

  const getInspection = async (id: string): Promise<void> => {
    dispatch({
      type: InspectionActionTypes.reset,
    });

    try {
      dispatch({
        type: InspectionActionTypes.loading,
        payload: {
          isLoading: true,
        },
      });
      const { data } = await inspectionService.getInspection({ id });

      dispatch({
        type: InspectionActionTypes.add,
        payload: {
          inspection: data,
        },
      });
    } catch (error) {
      dispatch({
        type: InspectionActionTypes.error,
      });
    }
  };

  return (
    <div>
      <Title>Search</Title>
      <SingleInputForm
        buttonText="SEARCH"
        handleClick={getInspection}
        placeholder="e.g.: hNrn6rCp"
      />
      <ResultList>
        {!isNil(inspection) && (
          <>
            <p>Urls:</p>
            {inspection?.urls.map((el) => (
              <ResultListItem key={el}>
                <a target="_blank" href={el} rel="noreferrer">
                  {el}
                </a>
              </ResultListItem>
            ))}
          </>
        )}
      </ResultList>
    </div>
  );
};
