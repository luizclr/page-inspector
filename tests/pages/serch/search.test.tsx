import { faker } from "@faker-js/faker";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import { Search } from "~/pages/search/search";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    state: { key: faker.random.word(), id: faker.random.alphaNumeric(8) },
  }),
}));

describe("<Search />", () => {
  it("should be able to mount <Search /> component", async () => {
    // given
    const { asFragment } = render(
      <BrowserRouter>
        <Search />
      </BrowserRouter>
    );

    // when
    const sut = asFragment();

    // then
    expect(sut).toMatchSnapshot();
  });
});
