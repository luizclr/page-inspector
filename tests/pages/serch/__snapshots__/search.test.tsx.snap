// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should be able to mount <Search /> component 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="sc-hLBbgP dCYQyq"
    >
      Search
    </h2>
    <div
      class="sc-bcXHqe ggCrYr"
    >
      <input
        class="sc-gswNZR EyzNS"
        placeholder="e.g.: hNrn6rCp"
        type="text"
        value=""
      />
      <button
        class="sc-dkrFOg jsUcRe"
        disabled=""
        type="submit"
      >
        SEARCH
      </button>
    </div>
    <ul
      class="sc-gKPRtg heGRtm"
    />
  </div>
</DocumentFragment>
`;
